sigar_linux.go

```diff
@@ -137,6 +137,161 @@ func (self *CpuList) Get() error {
 
 	return err
 }
+func (self *NetProtoV6Stats) Get() error {
+	return readFile(Procd+"/net/snmp6", func(line string) bool {
+		fields := strings.Fields(line)
+		if len(fields) != 2 {
+			return true
+		}
+		switch fields[0] {
+		case "Ip6InReceives":
+			self.IP.InReceives, _ = strtoull(fields[1])
+		case "Ip6InAddrErrors":
+			self.IP.InAddrErrors, _ = strtoull(fields[1])
+		case "Ip6OutForwDatagrams":
+			self.IP.ForwDatagrams, _ = strtoull(fields[1])
+		case "Ip6InDelivers":
+			self.IP.InDelivers, _ = strtoull(fields[1])
+		case "Ip6InDiscards":
+			self.IP.InDiscards, _ = strtoull(fields[1])
+		case "Ip6OutRequests":
+			self.IP.OutRequests, _ = strtoull(fields[1])
+		case "Icmp6InMsgs":
+			self.ICMP.InMsgs, _ = strtoull(fields[1])
+		case "Icmp6InErrors":
+			self.ICMP.InErrors, _ = strtoull(fields[1])
+		case "Icmp6InDestUnreachs":
+			self.ICMP.InDestUnreachs, _ = strtoull(fields[1])
+		case "Icmp6OutMsgs":
+			self.ICMP.OutMsgs, _ = strtoull(fields[1])
+		case "Icmp6OutDestUnreachs":
+			self.ICMP.OutDestUnreachs, _ = strtoull(fields[1])
+		case "Udp6InDatagrams":
+			self.UDP.InDatagrams, _ = strtoull(fields[1])
+		case "Udp6OutDatagrams":
+			self.UDP.OutDatagrams, _ = strtoull(fields[1])
+		case "Udp6InErrors":
+			self.UDP.InErrors, _ = strtoull(fields[1])
+		case "Udp6NoPorts":
+			self.UDP.NoPorts, _ = strtoull(fields[1])
+		}
+		return true
+	})
+}
+
+func (self *NetProtoV4Stats) Get() error {
+	// Each line starts with a header that describes the values, e.g.:
+	// Udp: InDatagrams NoPorts InErrors OutDatagrams RcvbufErrors SndbufErrors
+	// This map keeps track of the names of each position for each protocol. Reload
+	// it each time we parse.
+	protocols := make(map[string]map[int]string)
+
+	return readFile(Procd+"/net/snmp", func(line string) bool {
+		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			return true
+		}
+
+		// The first field is the protocol name
+		protocol := fields[0]
+
+		// The first line with this protocol name defines the column names
+		positions := protocols[protocol]
+
+		// Parse column headers if needed
+		if positions == nil {
+			positions = make(map[int]string)
```

Comment:
Should the mapping be from string to int? (`map[string]int`)?  Then the key is the field name and the value is the position.  That way, in the next section you can do something like this:

```
case "Ip:":
  self.IP.InReceives, _ = strtoull(fields[positions["InReceives"]])
  self.IP.InAddrErrors, _ = strtoull(fields[positions["InAddrErrors"]])
  ...
```

This eliminates the need for the `for position, field` loop as well as the `switch positions[position]` statement.  You can reference each value by name.